compose/utils.py

```diff
@@ -170,3 +170,12 @@ def truncate_id(value):
     if len(value) > 12:
         return value[:12]
     return value
+
+
+def unique_everseen(iterable, key=lambda x: x):
```

Comment:
Can you explain what the intent is for the `key` parameter? The only place you seem to use it is in the unit tests.

Reply:
And can we rename this to `remove_duplicates`? :innocent:

Reply:
Oh, I just realized that it's based on the function with the same name in https://github.com/erikrose/more-itertools ; name is fine, in that respect. The `key` param should be fixed to work correctly (see below) or removed if we don't think we'll use it.

Reply:
Fixed by 5b9b519.